core/src/main/scala/sttp/openai/OpenAIErrors.scala

```diff
@@ -0,0 +1,71 @@
+package sttp.openai
+
+import sttp.client4.HttpError
+
+object OpenAIErrors {
+  sealed abstract class OpenAIError(
```

Comment:
maybe let's make it simply an `OpenAIException`?

Reply:
But good idea to represent open ai errors as their own model :) Maybe we should use the same approach in `OpenAI` as well?